### ISSUE TITLE
[BUG] Incoming sound keeps playing even after the call is disconnected

### ISSUE THREAD

**Issue by kamalbennani** [twilio/twilio-voice.js#133]
- [x] I have verified that the issue occurs with the latest twilio.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-voice.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-voice.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/TwilioDevEd/voice-javascript-sdk-quickstart-node) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

```js
// receive an incoming call
this.device.on('incoming', (call: Call) => {
  // save call
  this._activeCall = call;

  // listen to call events
  this._activeCall.on('disconnect', () => console.log('call got disconnected'));
  this._activeCall.on('accept', () => console.log('call got answered'));
});

// pickup the call
this._activeCall.accept();

// right the way hung up before the call got time to connect (before pstream->answer event)
this._activeCall.disconnect();

// incoming sound keeps playing forever
```

**Expected behavior:**

The incoming sound should be stopped after a call disconnection

**Actual behavior:**

The incoming sound keeps playing forever

**Software versions:**

 - [x] Browser(s): Version 108.0.5359.124 (Official Build) (x86_64)
 - [x] Operating System: 12.5.1 Mac

**Solution**

I've debugged the situation and I found the following issue:

In the device class, the SDK only stops the incoming sound in the following situation:
- [accept](https://github.com/twilio/twilio-voice.js/blob/master/lib/twilio/device.ts#L1208)
- [cancel](https://github.com/twilio/twilio-voice.js/blob/master/lib/twilio/device.ts#L1036)
- [reject](https://github.com/twilio/twilio-voice.js/blob/master/lib/twilio/device.ts#L1054)
- [transportClose](https://github.com/twilio/twilio-voice.js/blob/master/lib/twilio/device.ts#L1069)
- [error](https://github.com/twilio/twilio-voice.js/blob/master/lib/twilio/device.ts#L1026)

But none of these events is sent in the case I described above.

The only event emitted is "disconnect" because the call got disconnected just after attempting to connect.

To resolve the issue, it's as simple as calling `this._maybeStopIncomingSound();` at the end of the [disconnect event](https://github.com/twilio/twilio-voice.js/blob/master/lib/twilio/device.ts#L1044)

I would like to hear your thoughts about this, to know if you are also able to reproduce it and if I should pull up a PR for it.

**Comment by mhuynh5757** [comment]
Hey @kamalbennani, thanks for filing this issue and providing a solution. We'll add it as soon as we can, I've filed an internal ticket to keep track of this.

**Comment by kamalbennani** [comment]
Hi @mhuynh5757 thank you for your quick response 🙇
Let me know if you want me to pull up a PR.
In all cases, I applied the patch to our current app and it's working perfectly fine.

**Comment by mhuynh5757** [comment]
Hey @kamalbennani, PRs are always welcome. Once the fix is done, our team will run through the usual regression testing before we can release. Thanks for the community contributions, we genuinely appreciate it.

**Comment by kamalbennani** [comment]
Hi @mhuynh5757 I've opened a [PR](https://github.com/twilio/twilio-voice.js/pull/134) for this. I'd like to have your thoughts on it.
Thanks.

**Comment by charliesantos** [comment]
should be fixed on 2.3.1

**Comment by bradleyy1012** [comment]
Is it possible that this bug has resurfaced after the latest update on chrome?
